src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,43 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.Arrays;
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private int[] array;
+    private int n;
 
     public MergeSortAction(int[] array) {
+        this.array = array;
+        this.n = array.length;
+    }
+
+    @Override
+    protected void compute() {
+        if (array.length == 1) {
+            return;
+        }
+        var left = Arrays.copyOfRange(array, 0 , n/2);
+        var right = Arrays.copyOfRange(array, n/2 , n);
```

Comment:
Similarly, use 'n' instead of 'array.length' for consistency and slight performance optimization.